url.bs

```diff
@@ -88,6 +88,334 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
+   <td>
+    <p>When an <a>opaque host</a> (in a URL that <a>is not special</a>) contains a
+    <a>forbidden host code point</a>.
+   <p class=example id=example-forbidden-host-code-point>"<code>foo://exa[mple.org</code>"
+   <td>✅
+ <tbody>
+  <tr>
+   <th colspan=3 scope=rowgroup><a href=#url-parsing>URL parsing</a>
+  <tr>
+   <td><dfn>unexpected-C0-control-or-space</dfn>
+   <td>
+    <p>The input to the <a>URL parser</a> contains a leading or trailing <a>C0 control or space</a>. The
+    URL parser subsequently strips any matching code points.
+    <p class=example id=example-unexpected-c0-control-or-space>"<code> https://example.org </code>"
+   <td>❌
+  <tr>
+   <td><dfn>unexpected-ASCII-tab-or-newline</dfn>
+   <td>
+    <p>The input to the URL parser contains <a>ASCII tab or newlines</a>. The URL parser
+    subsequently strips any matching code points.
+    <p class=example id=example-unexpected-ascii-tab-or-newline>"<code>ht<br>tps://example.org</code>"
+   <td>❌
+  <tr>
+   <td><dfn>invalid-scheme-start</dfn>
+   <td>
+    <p>The first code point of a URL's <a for=url>scheme</a> is not an <a>ASCII alpha</a>.
+    <p class=example id=example-invalid-scheme-start>"3ttps://example.org"
+   <td>✅
+  <tr>
+   <td><dfn>file-scheme-missing-following-solidus</dfn>
+   <td>
+    <p>The URL parser encounters a URL with a "<code>file</code>" <a for=url>scheme</a> that is not
+    followed by "<code>//</code>".
+    <p class=example id=example-file-scheme-missing-following-solidus>"<code>file:c:/my-secret-folder</code>"
+   <td>❌
+  <tr>
+   <td><dfn>invalid-scheme</dfn>
+   <td>
+    <p>The URL's <a for=url>scheme</a> contains an invalid code point.
+    <p class=example id=example-invalid-scheme>"<code>^_^://example.org</code>" and
+    "<code>https//example.org</code>"
+   <td>✅
+  <tr>
+   <td><dfn>missing-scheme-non-relative-URL</dfn>
+   <td>
+    <p>The input is missing a <a for=url>scheme</a>, because it does not begin with an
+    <a>ASCII alpha</a>, and either no <a>base URL</a> was provided or the <a>base URL</a> cannot be
+    used as a <a>base URL</a> because it has an <a for=url>opaque path</a>.
+    <div class=example id=example-missing-scheme-non-relative-url>
+     <p>Input's <a for=url>scheme</a> is missing and no <a>base URL</a> is given:
+     <pre><code class=lang-javascript>
+let url = new URL("💩");</code></pre>
+     <p>Input's <a for=url>scheme</a> is missing, but the <a>base URL</a> has an
+     <a for=url>opaque path</a>.
+     <pre><code class=lang-javascript>
+let url = new URL("💩", "mailto:user@example.org");</code></pre>
+    </div>
+   <td>✅
+  <tr>
+   <td><dfn>relative-URL-missing-beginning-solidus</dfn>
+   <td>
+    <p>The input is a <a>relative-URL String</a> that does not begin with U+002F (/).
+    <pre class=example id=example-relative-url-missing-beginning-solidus><code class="lang-javascript">
+let url = new URL("foo.html", "https://example.org/");</code></pre>
```

Comment:
Why the heck would this be an error? Does this mean `<a href="foo.html">` is an error?

Reply:
Good point. I think this would only be hit by `new URL("https:foo.html", "https://example.com/")`, but that clearly requires a better name.